### DISCUSSION TITLE
How to set `buffer_size` for Monolog loggers using Laravel configuration?

### DISCUSSION THREAD

**Discussion by plumthedev** [laravel/framework#52659] [Q&A]
Based on [Bug Report on OpenSearch PHP LIbrary](https://github.com/opensearch-project/opensearch-php/issues/178) I wanted to ask if there is a way to set `buffer_size` for Monolog loggers which are used by Laravel

**Comment by TomKeur** [comment]
I'm facing the same issue, we're running long PHP jobs (queue:worker), after a few hours the job would use a lot of memory and got killed by Out of Memory Killer on Linux.

For now I've fixed it with a patch via `[cweagans/](https://packagist.org/packages/cweagans/)composer-patches`
https://packagist.org/packages/cweagans/composer-patches

My `laravel-log-manager-buffer-size.patch` file:


```
Index: vendor/laravel/framework/src/Illuminate/Log/LogManager.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/vendor/laravel/framework/src/Illuminate/Log/LogManager.php b/vendor/laravel/framework/src/Illuminate/Log/LogManager.php
--- a/vendor/laravel/framework/src/Illuminate/Log/LogManager.php
+++ b/vendor/laravel/framework/src/Illuminate/Log/LogManager.php        (date 1732198474431)
@@ -464,8 +464,8 @@
             $handler = new FingersCrossedHandler(
                 $handler,
                 $this->actionLevel($config),
-                0,
-                true,
+                $config['buffer_size'] ?? 0,
+                $config['bubble'] ?? true,
                 $config['stop_buffering'] ?? true
             );
         }
```

The memory issue is gone, I'm sure I've I can create a pull request, to make these `buffer_size` and `bubble` configurable.
